src/sql/src/query.rs

```diff
@@ -1799,9 +1846,9 @@ fn plan_aggregate(ecx: &ExprContext, sql_func: &Function) -> Result<AggregateExp
     let (mut expr, mut func) = match (name.as_str(), &sql_func.args) {
         // COUNT(*) is a special case that doesn't compose well
         ("count", FunctionArgs::Star) => (
-            // Ok to use `ScalarType::Unknown` here because this expression
-            // can't ever escape the surrounding reduce.
-            ScalarExpr::literal_null(ScalarType::Unknown),
+            // The scalar type of the null doesn't matter because this
+            // expression can't ever escape the surrounding reduce.
```

Comment:
nit: s/can't ever escape/can't escape/ ...or something grammatically equivalent

Reply:
Done, but I"m not sure I see what was wrong with the original?